src/Aside/Aside.js

```diff
@@ -3,63 +3,21 @@ import Controls from '../Controls/Controls.js';
 import AsideBody from '../AsideBody/AsideBody.js';
 import './Aside.css';
 
-class Aside extends Component {
-  constructor(props) {
-    super(props);
-    let storedCardNames = [];
-    if (props.userCardsData.length > 0) {
-      storedCardNames = props.userCardsData.map(card => card.cardName);
-    }
-    this.state = {
-      cardNames: storedCardNames,
```

Comment:
Having the `cardNames` property here in Aside's state was the root of a lot of complications. Shifting the responsibility for error handling to the Controls component was a large part of removing the need for `cardNames`.